### PR TITLE
Filling out missing members of System.Linq.Expressions namespace

### LINKED ISSUE #13483: Stragglers in S.Linq.Expressions
These are in NS2.0 but not exposed

```
    public sealed class Expression<TDelegate> : LambdaExpression {
        public new TDelegate Compile(DebugInfoGenerator debugInfoGenerator);
    }
    public abstract class ExpressionVisitor {
        protected internal virtual Expression VisitDynamic(DynamicExpression node);
    }
    public abstract class LambdaExpression : Expression {
        public Delegate Compile(DebugInfoGenerator debugInfoGenerator);
    }
    public abstract class MemberBinding {
        protected MemberBinding(MemberBindingType type, MemberInfo member);
    }
```

### PR DESCRIPTION
**Pull request by joperezr**
Fixes #13483 #13332 

Filling out missing members of System.Linq.Expressions namespace.

### CODE REVIEW

**Comment by joperezr** [comment]
cc: @weshaggard @danmosemsft @AlexGhiondea

**Comment by stephentoub** [comment]
cc: @VSadov, @bartdesmet

File: src/System.Linq.Expressions/src/System/Runtime/CompilerServices/DebugInfoGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    public abstract class DebugInfoGenerator
+    {
+        /// <summary>
+        /// Creates PDB symbol generator.
+        /// </summary>
+        /// <returns>PDB symbol generator.</returns>
+        public static DebugInfoGenerator CreatePdbGenerator()
+        {
+            // Creating PDBs is not suported in .NET Core
+            throw new NotSupportedException();
```
Comment (danmosemsft) [inline src/System.Linq.Expressions/src/System/Runtime/CompilerServices/DebugInfoGenerator.cs]: >NotSupportedException [](start = 22, length = 21)

PlatformNotSupportedException ?

File: src/System.Linq.Expressions/ref/System.Linq.Expressions.cs [hunk 2]
```diff
@@ -358,8 +358,7 @@ public partial class DynamicExpression : System.Linq.Expressions.Expression, Sys
     public abstract partial class DynamicExpressionVisitor : System.Linq.Expressions.ExpressionVisitor
     {
         protected DynamicExpressionVisitor() { }
-        // Member is excluded in the base so it is automatically excluded for this type without any way to override that behavior
-        protected virtual Expression VisitDynamic(DynamicExpression node) { throw null; }
+        protected internal virtual new System.Linq.Expressions.Expression VisitDynamic(System.Linq.Expressions.DynamicExpression node) { throw null; }
```
Comment (danmosemsft) [inline src/System.Linq.Expressions/ref/System.Linq.Expressions.cs]: >new [](start = 35, length = 3)

Why new? This is override in DEsktop.

**Comment by danmosemsft** [comment]
thoughts about unit tests? It looks like you didn't have any in your previous change. I'm guessing it would need a lot of familiarity with Linq to write some.

**Comment by danmosemsft** [comment]
:shipit:

File: src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.netstandard1.7.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Dynamic.Utils;
+using System.Reflection;
```
Comment (justinvp) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.netstandard1.7.cs]: Nit: "Remove and Sort"

File: src/System.Linq.Expressions/ref/System.Linq.Expressions.cs (hunk 2 above)
Comment (joperezr) [inline src/System.Linq.Expressions/ref/System.Linq.Expressions.cs]: You are right, I'll fix this.

File: src/System.Linq.Expressions/src/System/Runtime/CompilerServices/DebugInfoGenerator.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    public abstract class DebugInfoGenerator
+    {
+        /// <summary>
+        /// Creates PDB symbol generator.
+        /// </summary>
+        /// <returns>PDB symbol generator.</returns>
+        public static DebugInfoGenerator CreatePdbGenerator()
+        {
+            // Creating PDBs is not suported in .NET Core
+            throw new PlatformNotSupportedException();
```
Comment (JonHanna) [inline src/System.Linq.Expressions/src/System/Runtime/CompilerServices/DebugInfoGenerator.cs]: Is this not supported yet, not supported unless it gets screamed for and/or implemented by the interested or never to be supported?

**Pull request merged**